### DISCUSSION TITLE
Enabling GitHub Pro for Private GitHub Pages Site with Public Visibility

### DISCUSSION THREAD

**Discussion by ccsekhar20** [community/community#83079] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello,

I currently have a GitHub Pages site that I'd like to make private to protect the source code, but I still want the site to be visible to others. After exploring GitHub Pro, I'm unsure about the steps to achieve this.

Here are my main questions:

Is it possible to use GitHub Pro to make my GitHub Pages site private while keeping it accessible to the public?
What are the specific steps or settings I need to configure to achieve this?
Do you have any suggestions or best practices I should know when using GitHub Pro in this context?

I appreciate any guidance or insights the community can provide on this matter. Thank you!

**Comment by queenofcorgis** [comment, marked as answer]
Hi there @ccsekhar20 and Happy New Year! 

> Is it possible to use GitHub Pro to make my GitHub Pages site private while keeping it accessible to the public?

Yes, you would make your repository used to host the site private, but the site would still be publicly available on the internet. 

> What are the specific steps or settings I need to configure to achieve this?

You would need to change the repository's visibility: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/setting-repository-visibility#changing-a-repositorys-visibility - this doc will give you the instructions and then proceed with building your Pages site as usual. 

> Do you have any suggestions or best practices I should know when using GitHub Pro in this context?

I don't have anything specific as this should be fairly straightforward, but I'd love to hear from any other members about their experiences. 

Cheers and good luck!

**Reply by ccsekhar20** [reply]
Happy New Year to you too! Thank you so much for answering my questions.
